src/main/kotlin/no/nav/familie/ef/iverksett/brev/DistribuerVedtaksbrevTask.kt

```diff
@@ -17,17 +18,39 @@ import java.util.UUID
                      triggerTidVedFeilISekunder = 15 * 60L,
                      beskrivelse = "Distribuerer vedtaksbrev.")
 class DistribuerVedtaksbrevTask(private val journalpostClient: JournalpostClient,
-                                private val tilstandRepository: TilstandRepository) : AsyncTaskStep {
+                                private val tilstandRepository: TilstandRepository,
+                                private val featureToggleService: FeatureToggleService) : AsyncTaskStep {
 
     private val logger: Logger = LoggerFactory.getLogger(this::class.java)
 
     override fun doTask(task: Task) {
         val behandlingId = UUID.fromString(task.payload)
-        val journalpostId = tilstandRepository.hentJournalpostResultat(behandlingId)?.journalpostId
-        val bestillingId = journalpostId?.let { journalpostClient.distribuerBrev(it) }
-        tilstandRepository.oppdaterDistribuerVedtaksbrevResultat(behandlingId = behandlingId,
-                                                                 DistribuerVedtaksbrevResultat(bestillingId = bestillingId)
-        )
+
+        validerJournalpostresultat(behandlingId)
+
+        val journalpostResultat = tilstandRepository.hentJournalpostResultat(behandlingId)
+
+        val distribuerteJournalposter = tilstandRepository.hentdistribuerVedtaksbrevResultat(behandlingId)?.keys ?: emptySet()
+
+        journalpostResultat?.filter { (_, journalpostResultat) ->
+            journalpostResultat.journalpostId !in distribuerteJournalposter
+        }?.forEach { (_, journalpostResultat) ->
+            val bestillingId = journalpostClient.distribuerBrev(journalpostResultat.journalpostId)
```

Comment:
Denne er også fin - for her kan vi potensielt få inn en fremtidig: "Skal distribuere vedtaksbrev". Det kommer nok til et punkt hvor vedtaksbrev/infobrev ikke nødvendigvis skal distribueres, men jeg er overbevist om at vi må journalføre _noe_ :)